src/asyncresults.jl

```diff
@@ -0,0 +1,256 @@
+"An asynchronous PostgreSQL query"
+mutable struct AsyncResult
+    "The LibPQ.jl Connection used for the query"
+    jl_conn::Connection
+
+    "Keyword arguments to pass to Result on creation"
+    result_kwargs::Ref
+
+    "Whether or not the query should be cancelled, if running"
+    should_cancel::Bool
+
+    "Task which errors or returns a LibPQ.jl Result which is created once available"
+    result_task::Task
+
+    function AsyncResult(jl_conn::Connection, result_kwargs::Ref)
+        async_result = new(jl_conn, result_kwargs, false)
+        jl_conn.async_result = async_result
```

Comment:
should we have:
```
jl_conn.async_result ==nothing || throw(ArgumentError("Connection already serving another async request"))
```
Or an assertion to the same effect?

Reply:
Now it's okay if the connection is serving another async result when it gets created.